app/assets/javascripts/controllers/floating_ip/floating_ip_form_controller.js

```diff
@@ -1,34 +1,42 @@
-ManageIQ.angular.app.controller('floatingIpFormController', ['$http', '$scope', 'floatingIpFormId', 'miqService', function($http, $scope, floatingIpFormId, miqService) {
+ManageIQ.angular.app.controller('floatingIpFormController', ['floatingIpFormId', 'miqService', 'API', function(floatingIpFormId, miqService, API) {
   var vm = this;
-  vm.floatingIpModel = { floating_ip_address: '' };
-  vm.formId = floatingIpFormId;
-  vm.afterGet = false;
-  vm.modelCopy = angular.copy( vm.floatingIpModel );
-  vm.model = "floatingIpModel";
 
-  ManageIQ.angular.scope = vm;
-  vm.saveable = miqService.saveable;
+  var init = function() {
+    vm.afterGet = false;
 
-  if (floatingIpFormId == 'new') {
-    vm.floatingIpModel.floating_ip_address = "";
-    vm.floatingIpModel.description = "";
-    vm.newRecord = true;
-  } else {
-    miqService.sparkleOn();
+    vm.floatingIpModel = { address: null };
+    vm.available_ems = null;
 
-    $http.get('/floating_ip/floating_ip_form_fields/' + floatingIpFormId)
-      .then(getFloatingIpFormData)
-      .catch(miqService.handleFailure);
-  }
+    vm.formId = floatingIpFormId;
+    vm.model = "floatingIpModel";
+    vm.newRecord = floatingIpFormId === "new";
+    vm.saveable = miqService.saveable;
+
+    if (vm.newRecord) {
+      vm.afterGet = true;
```

Comment:
This should be in `.then` like it is in `else` below. Otherwise form will be displayed before API call is resolved.

Reply:
You're right I forgot that flag triggers the form 👍. Done